.jenkinsci/bindings.groovy

```diff
@@ -1,27 +1,69 @@
 #!/usr/bin/env groovy
 
-def doBindings() {
-  def cmake_options = ""
-  if (params.JavaBindings) {
-    cmake_options += " -DSWIG_JAVA=ON "
-  }
-  if (params.PythonBindings) {
-    cmake_options += " -DSWIG_PYTHON=ON "
-  }
-  // In case language specific options were not set,
-  // build for each language
-  if (!params.JavaBindings && !params.PythonBindings) {
-    cmake_options += " -DSWIG_JAVA=ON -DSWIG_PYTHON=ON "
-  }
+def doJavaBindings(buildType=Release) {
+  def currentPath = sh(script: "pwd", returnStdout: true).trim()
+  def commit = env.GIT_COMMIT
+  def artifactsPath = sprintf('%1$s/java-bindings-%2$s-%3$s-%4$s.zip', 
+    [currentPath, buildType, sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
   sh """
     cmake \
       -H. \
       -Bbuild \
-      -DCMAKE_BUILD_TYPE=Release \
-      ${cmake_options}
+      -DCMAKE_BUILD_TYPE=$buildType \
+      -DSWIG_JAVA=ON
+  """
+  sh "cd build; make -j${params.PARALLELISM} irohajava"
+  sh "zip -j $artifactsPath build/shared_model/bindings/*.java build/shared_model/bindings/libirohajava.so"
+  sh "cp $artifactsPath /tmp/bindings-artifact"
+  return artifactsPath
+}
+
+def doPythonBindings(buildType=Release) {
+  def currentPath = sh(script: "pwd", returnStdout: true).trim()
+  def commit = env.GIT_COMMIT
+  def artifactsPath = sprintf('%1$s/python-bindings-%2$s-%3$s-%4$s.zip', 
+    [currentPath, buildType, sh(script: 'date "+%Y%m%d"', returnStdout: true).trim(), commit.substring(0,6)])
+  sh """
+    cmake \
+      -H. \
+      -Bbuild \
+      -DCMAKE_BUILD_TYPE=$buildType \
+      -DSWIG_PYTHON=ON \
+      -DSUPPORT_PYTHON2=ON
```

Comment:
The function relates only to the python2 bindings, so maybe rename it accordingly?

Reply:
OK, as far as I see current version of CMakeLists file allows to specify only one version of Python that will be supported with generated bindings:
https://github.com/hyperledger/iroha/blob/61969d20d1e8d71c60f4417fe271c30cf2d6dce4/shared_model/bindings/CMakeLists.txt#L47-L51
Am I right? Can we modify it to support Python3 by default, while `SUPPORT_PYTHON2` would really enable support for Python2 and not just switching Python version?

Reply:
Yeah, I think it can be done, though directories would change

Reply:
We can run cmake twice: one time with ```-DSUPPORT_PYTHON2=ON``` and another with ```-DSUPPORT_PYTHON2=OFF```.